### ISSUE TITLE
Add Jakarta support

### ISSUE THREAD

**Issue by forgedhallpass** [OWASP/www-project-csrfguard#176]
There is a new branch named [`jakarta`](https://github.com/OWASP/www-project-csrfguard/tree/jakarta) created for this purpose, but the preferred solution would be to use some automated approach (e.g. [Eclipse Transformer](https://github.com/eclipse/transformer)) that would do the required changes as a release step, hence removing the need for maintaining two almost identical branches.

Related discussions:
* https://github.com/OWASP/www-project-csrfguard/discussions/41
* https://github.com/OWASP/www-project-csrfguard/discussions/131

**Comment by stefano-1973** [comment]
Looking to the other projects, they give javax.* support in the old/current version and in the new major version give only jakarta.* support. 
For example for Jetty and similarly for other projects:  (https://www.eclipse.org/jetty/)
```
Jetty Version         Servlet      License                                  Java
Jetty 9.4.x             3.1        JavaEE 8/javax.servlet.*                 Java 8
Jetty 10.0.x            4.0        JakartaEE 8/javax.servlet.*              Java 11+
Jetty 11.0.x            5.0        JakartaEE 9/jakartaee.servlet.*          Java 11+
```
So you can leave javax.* at the old and current 3.x/4.x versions and create a new 5.x version with jakarta.* support only: the only version where new developments will be. So the master branch will only support Jakarta.* . (At most, if there is some serious problem for users using javax.*, you can create a new 4.x.x release version with only the specific fix, nothing more).

This way you won't need to maintain two almost identical branches (then over time fewer and fewer people will use javax.*)

**Comment by stefano-1973** [comment]
What do you think about my proposal above? Am I forgetting something important?

**Comment by forgedhallpass** [comment]
> At most, if there is some serious problem for users using javax.* , you can create a new 4.x.x release version with only the specific fix, nothing more.

This actually means that we'd need to maintain two branches. That is why I was thinking that maybe we could use something like Eclipse Transformer to make this transformation automatic (changing the dependency and the imports).

I can't comment on how many people uses `jakarta` over `javax`, but until this point there were only the two requests (see above) to provide support.

**Comment by stefano-1973** [comment]
In the meantime, better to have the need to update two branches than to have nothing. 
For example, there is now no Jakarta resource in the maven general public repository for use in maven projects. Anyone switching to Jakarta is not facilitated to use this library...

P.S.: not everyone that switch to Jakarta create a "request"... There are many Jakarta EE 9 users already and more to come (read https://eclipse-foundation.blog/2022/09/26/survey-says-confidence-continues-to-grow-in-the-jakarta-ee-ecosystem/)

**Comment by forgedhallpass** [comment]
@stefano-1973 If you fix the build in the `jakarta` branch and make sure the test application is deployable, then I'll do a `-jakarta` release.

The `tomcat7-maven-plugin` probably will have to be changed to a newer `cargo-maven3-plugin` and the container should be `tomcat10x` or newer.

**Comment by stefano-1973** [comment]
Your requests:

1) "fix the build in the jackarta branch"
2) "make sure the "test application" is deployable"

My related questions/consideration:

1) What is the problem with the branch? My original pull-request was buildable and the generated library usable
2) I don't know this "test application", how to use it, what it does. I'm sorry but at the moment I don't have time to study, configure, correct, install and debug the final result of the"test application"

**Comment by forgedhallpass** [comment]
> What is the problem with the branch? My original pull-request was buildable and the generated library usable

We use the [test-jsp](https://github.com/OWASP/www-project-csrfguard/tree/master/csrfguard-test/csrfguard-test-jsp) app to test the solution and to make sure that all the functionalities are working. Ideally integration tests would have already failed the build, but we don't have those yet, hence the testing is manual.

> I don't know this "test-application", how to use it, what it does. I'm sorry but at the moment I don't have time to study, configure, correct, install and debug the final result of the "test-application"

I understand that, but until the changes are not integrated into the test app, the ticket is not considered finished, hence it will not be released. The steps on how to build and start the application are [documented](https://github.com/OWASP/www-project-csrfguard#building-the-code) in the readme. 

If nobody contributes the required changes, I'll do it myself, but my schedule is quite busy, so it will take some time.

**Comment by rzanner** [comment]
Any news on this? Just preparing the update to the newest jakarta libraries, only CsrfGuard is preventing the switch now... :-/

**Comment by eitch** [comment]
Is the test-application a unit test?

I used @forgedhallpass branch, and did a few tweaks: https://github.com/eitch/www-project-csrfguard

I was able to start the test WAR and play around with it in the browser. It looked fine. But i think an original dev should probably take a look and test it too. The work has been done by @forgedhallpass and i think all is fine.

**Comment by forgedhallpass** [comment]
@rzanner, @stefano-1973 I've tested the bundled application with the `jakarta` branch, using `Apache Tomcat 10.1.9` and `OpenJDK 11.0.19`, but the JSP support is not working. I've tried changing the `taglib` in the TLD to version `3.0`, which solved one of the initial issues, but the automatic TLD discovery is still not working:

```java
org.apache.jasper.JasperException: The absolute uri: [https://owasp.org/www-project-csrfguard/Owasp.CsrfGuard.tld] cannot be resolved in either web.xml or the jar files deployed with this application
	org.apache.jasper.compiler.DefaultErrorHandler.jspError(DefaultErrorHandler.java:54)
	org.apache.jasper.compiler.ErrorDispatcher.dispatch(ErrorDispatcher.java:294)
	org.apache.jasper.compiler.ErrorDispatcher.jspError(ErrorDispatcher.java:81)
	org.apache.jasper.compiler.TagLibraryInfoImpl.generateTldResourcePath(TagLibraryInfoImpl.java:251)
	org.apache.jasper.compiler.TagLibraryInfoImpl.<init>(TagLibraryInfoImpl.java:122)
	org.apache.jasper.compiler.Parser.parseTaglibDirective(Parser.java:429)
	org.apache.jasper.compiler.Parser.parseDirective(Parser.java:487)
	org.apache.jasper.compiler.Parser.parseElements(Parser.java:1444)
	org.apache.jasper.compiler.Parser.parse(Parser.java:138)
	org.apache.jasper.compiler.ParserController.doParse(ParserController.java:245)
	org.apache.jasper.compiler.ParserController.parse(ParserController.java:106)
	org.apache.jasper.compiler.Compiler.generateJava(Compiler.java:211)
	org.apache.jasper.compiler.Compiler.compile(Compiler.java:396)
	org.apache.jasper.compiler.Compiler.compile(Compiler.java:372)
	org.apache.jasper.compiler.Compiler.compile(Compiler.java:356)
	org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
	org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
	org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:380)
	org.apache.jasper.servlet.JspServlet.service(JspServlet.java:328)
	jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
	org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	org.owasp.csrfguard.CsrfGuardFilter.handleNoSession(CsrfGuardFilter.java:122)
	org.owasp.csrfguard.CsrfGuardFilter.doFilter(CsrfGuardFilter.java:94)
	org.owasp.csrfguard.CsrfGuardFilter.doFilter(CsrfGuardFilter.java:68)
```

@eitch the application is not a unit test per se, but it is used to manually validate the functionalities provided by the solution.

**Comment by forgedhallpass** [comment]
Released under version [4.3.0-jakarta](https://github.com/OWASP/www-project-csrfguard/releases/tag/4.3.0-jakarta).
